### PR TITLE
generator-component@v2.0.0 - v2 release, more config and better structure

### PR DESCRIPTION
**Pull request by ashleynolan**
### Added
- `.yo-rc.json` to signify the destination root of the generator (to use the base of the mono-repo as our root directory, wherever the generator is run).
- Generator will now generate to a destination based on the users answers. For example, if it's specified that the component is an atom, the component will be generated into the `components/atoms` directory.
- Welcome prompt and update prompt which displays if the generator being run is out-of-date with the current published version.
- Option to exclude translations/`i18n` config when component doesn't need these.
- Excludes a number of directories and code when generated component is a service.

### Fixed
- `yeoman-generator` added to dependencies so that local testing doesn't error on missing dependency.

### Changed
- Structure of generator tidied up. Now uses `initializing` and `configuring` contexts rather than having all logic in the `writing` context.
- Prompts moved out into separate file.
- `_tests` folder updated (previously `tests`).
- `scripts` in `package.json` updated inline with recent changes to webdriver config.
- Unnecessary `scripts` removed based on yeoman options chosen.
- Sass loader config updated.

## UI Review Checks

Screen Shot 2021-01-28 at 10 58 29

- [x] README and/or UI Documentation has been updated

### CODE REVIEW

**Review by xander-marjoram — DISMISSED** [review]

File: packages/tools/generator-component/generators/app/index.js [hunk 1]
```diff
@@ -1,80 +1,111 @@
...
-            {
-                message: 'Does the component interact with any API\'s?',
-                name: 'needsTestingApiMocks',
-                type: 'confirm',
-                default: false
-            }
-        ]);
+    async initializing () {
+        const generatorWelcome = chalk`
+{yellow  ____  __  ____  ____  __  ____ }
```
Comment (DamianMullins) [inline packages/tools/generator-component/generators/app/index.js:14]: 💘

File: packages/tools/generator-component/package.json [hunk 2]
```diff
@@ -34,7 +34,9 @@
   ],
   "dependencies": {
     "gulp-rename": "1.2.2",
-    "lodash": "4.17.19"
+    "lodash": "4.17.19",
+    "update-notifier": "^5.0.1",
```
Comment (DamianMullins) [inline packages/tools/generator-component/package.json]: Pin these dependencies.

Comment (ashleynolan) [reply packages/tools/generator-component/package.json]: Done

File: packages/tools/generator-component/generators/app/templates/__package__.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
   "version": "0.1.0",
   "main": "dist/f-<%= name.default %>.umd.min.js",
   "files": [
-    "dist"<% if(needsComponentTests) { %>,
-    "test-utils"<% } %> <% if(needsTestingApiMocks) { %>,
+    "dist"<% if(config.needsComponentTests) { %>,
+    "test-utils"<% } %> <% if(config.needsTestingApiMocks) { %>,
     "src/services"<% } %>
   ],
   "homepage": "https://github.com/justeat/fozzie-components/tree/master/packages/f-<%= name.default %>",
```
Comment (DamianMullins) [inline packages/tools/generator-component/generators/app/templates/__package__.json]: This field needs to be updated to match the new project structure.

Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/__package__.json]: Fixed

**Review by kevinrodrigues — DISMISSED** [review]

File: packages/tools/generator-component/generators/app/templates/__package__.json [hunk 4]
```diff
@@ -30,28 +30,28 @@
     "prepublishOnly": "yarn lint && yarn test && yarn build",
     "build": "vue-cli-service build --target lib --name f-<%= name.default %> ./src/index.js",
     "lint": "vue-cli-service lint",
-    "lint:fix": "yarn lint --fix",
-    "lint:style": "vue-cli-service lint:style",
-    "report": "cd ../.. && yarn report",
-    "test-component:chrome": "wdio ../../../../wdio.conf.js --suite component",
+    "lint:fix": "yarn lint --fix"<% if(config.isComponent) { %>,
+    "lint:style": "vue-cli-service lint:style"<% } %>,
+    "report": "cd ../.. && yarn report"<% if(config.needsComponentTests) { %>,
```
Comment (matthewhardern) [inline packages/tools/generator-component/generators/app/templates/__package__.json]: missing `"test": "vue-cli-service test:unit"`

File: packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js [hunk 5]
```diff
@@ -1,6 +1,12 @@
 import <%= name.filename %>Component from '../../../test-utils/component-objects/f-<%= name.class %>.component';
 
 describe('f-<%= name.class %> component tests', () => {
+    beforeEach(() => {
+        browser.url('?path=<%= storybook.path %>');
```
Comment (matthewhardern) [inline packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js:5]: This is currently coming out in camel case when it should be hyphenated to match the generated storybook page

**Review by matthewhardern — CHANGES_REQUESTED** [review]

File: packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js (hunk 5 above)
Comment (matthewhardern) [reply packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js:5]: `?path=/story/components-molecules--media-element-component` vs `?path=story/components-molecules--mediaElement`

Comment (matthewhardern) [reply packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js:5]: first is the storybook page

File: packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue [hunk 6]
```diff
@@ -1,36 +1,36 @@
 <template>
     <div
-        :data-theme="theme"
         :class="$style['c-<%= name.class %>']"
-        data-test-id='<%= name.class %>-component'>
+        data-test-id="<%= name.class %>">
         {{ copy.text }}
```
Comment (matthewhardern) [inline packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue]: copy does not exist if you have not selected i18n

File: packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue [hunk 7]
```diff
@@ -1,36 +1,36 @@
 <template>
     <div
-        :data-theme="theme"
         :class="$style['c-<%= name.class %>']"
-        data-test-id='<%= name.class %>-component'>
```
Comment (matthewhardern) [inline packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue:5]: This change breaks the webdriver io tests

File: packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue [hunk 8]
```diff
@@ -1,36 +1,36 @@
...
     components: {},
     props: {
-        locale: {
+<% if (config.needsComponentTranslations) { %>        locale: {
             type: String,
             default: ''
         }
-    },
+<%} %>    },
     data () {
```
Comment (matthewhardern) [inline packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue:28]: could be a case where data has no return value if i18n is not selected

File: packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js (hunk 5 above)
Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js:5]: 👍 Will fix

File: packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue (hunk 6 above)
Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue]: Good spot – will update

File: packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue (hunk 7 above)
Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue:5]: Will update the tests

File: packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue (hunk 6 above)
Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue]: Fixed

File: packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js (hunk 5 above)
Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js:5]: Done

File: packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue (hunk 7 above)
Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue:5]: Done

File: packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue (hunk 8 above)
Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/src/components/Skeleton.vue:28]: Done

File: packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js (hunk 5 above)
Comment (matthewhardern) [reply packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js:5]: still not quite working `?path=story/components-molecules--content-banner` should be `?path=/story/components-molecules--content-banner-component`. Other than this happy to merge

Comment (ashleynolan) [reply packages/tools/generator-component/generators/app/templates/test/specs/component/f-skeleton.component.__spec__.js:5]: Have fixed this now

**Review by dandel10n — APPROVED** [review]

**Review by matthewhardern — APPROVED** [review]

**Pull request merged**